components/prize_pool/commons/prize_pool.lua

```diff
@@ -590,6 +590,21 @@ function PrizePool._getTournamentDate()
 	return Variables.varDefaultMulti('tournament_enddate', 'tournament_edate', 'edate', TODAY)
 end
 
+--- Formats a number for display as currency
+function PrizePool._formatPrize(value)
+	value = Math.round{value, 2}
+
+	-- if number is non integer and has only 1 digit after the `.` we need to fill it up
+	-- e.g. .5 --> .50
+	local decimal = string.match(value, '%.(.-)$') or ''
+	if String.isNotEmpty(decimal) then
+		decimal = '.' .. decimal .. string.rep('0', 2 - string.len(decimal))
+		value = math.floor(value)
+	end
+
+	return LANG:formatNum(value) .. decimal
```

Comment:
```suggestion
	local _, decimal = math.modf(value)
	if decimal == 0 then
		return LANG:formatNum(value)
	end
	return LANG:formatNum(tonumber(string.format('%.2f', value)))
```

Reply:
the tonumber before the format num kicks the `0` at the end again
![Screenshot 2022-07-05 12 51 17](https://user-images.githubusercontent.com/75081997/177312102-3a4b12cc-b03b-485f-bf94-ffeaf1cd8333.png)

Reply:
this would work:
```suggestion
	value = Math.round{value, 2}
	local _, decimal = math.modf(value)
	if decimal == 0 then
		return LANG:formatNum(value)
	end
	decimal = tonumber(string.format('%.2f', decimal))
	return LANG:formatNum(tonumber(string.format('%.2f', value))) .. string.rep('0', 4 - string.len(decimal))
```

Reply:
```
	decimal = tonumber(string.format('%.2f', decimal))
	return LANG:formatNum(value) .. string.rep('0', 4 - string.len(decimal))
```
Can be changed to
```
	return LANG:formatNum(value) .. string.format('%.2f', decimal)
```

Reply:
Also `value = Math.round{value, 2}` is redundant here right?

Reply:
> ```
> 	decimal = tonumber(string.format('%.2f', decimal))
> 	return LANG:formatNum(value) .. string.rep('0', 4 - string.len(decimal))
> ```
> 
> Can be changed to this right?
> 
> ```
> 	return LANG:formatNum(value) .. string.format('%.2f', decimal)
> ```

no
1) `string.format('%.2f', decimal)` contains the `0.`
2) the `value` already has the digits after the `.`

`string.rep('0', 4 - string.len(decimal))` only appends `0`s

could use
`return LANG:formatNum(value) .. string.rep('0', 4 - string.len(Math.round{decimal, 2}))`
though (`round` instead of `string.format` plus `tonumber` & push it into 1 line)

Reply:
another alternative is
```lua
	local integer, decimal = math.modf(value)
	if decimal == 0 then
		return LANG:formatNum(integer)
	end

	return LANG:formatNum(integer) .. string.format('%.2f', Math.round{decimal, 2})
```

Reply:
> the value already has the digits after the .

If you get the first return of `modf`, that's the integer part only.

Reply:
> > the value already has the digits after the .
> 
> If you get the first return of `modf`, that's the integer part only.

ik, but that still leaves the issue with the `0.` of the `string.format`
i don't remember a way to string.format to have right 0-padding without showing the `0.`
if that exists we could use it and would only be a 0.01 wrong maximum (rounding vs string.format that only cuts off i.e. rounds down)

Reply:
This should work right?

```suggestion``
	local roundedValue = Math.round{value, 2}
	local integer, decimal = math.modf(roundedValue )
	if decimal == 0 then
		return LANG:formatNum(integer)
	end
	return LANG:formatNum(integer) .. string.format('%.2f', decimal):sub(2)
```